src/agent.rs

```diff
@@ -103,35 +114,38 @@ pub enum TLSVersion {
     Unknown,
 }
 
-impl From<i32> for TLSVersion  {
+impl From<i32> for TLSVersion {
     fn from(value: i32) -> Self {
-       match value {
-           0x303 => TLSVersion::V1_2,
-           0x304 => TLSVersion::V1_3,
-           _ => TLSVersion::Unknown
-       }
+        match value {
+            0x303 => TLSVersion::V1_2,
+            0x304 => TLSVersion::V1_3,
+            _ => TLSVersion::Unknown,
+        }
     }
 }
 
 /// An [`Agent`] holds a non-cloneable reference to a Stream.
 pub struct Agent {
     pub descriptor: AgentDescriptor,
-    pub stream: OpenSSLStream,
+    pub stream: Box<dyn PUT>,
 }
 
 impl Agent {
-    pub fn new_openssl(
+    pub fn new<P: 'static + PUT>(
```

Comment:
Minor:
```suggestion
    pub fn new<P: PUT>(
```

I think its simpler to add the `'static` to the `PUT` trait.